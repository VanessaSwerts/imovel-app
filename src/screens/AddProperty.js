import React, { useState } from 'react'
import {
  Platform,
  SafeAreaView,
  FlatList,
  KeyboardAvoidingView,
  StyleSheet,
  Text,
  View,
  ScrollView,
  Image,
  TouchableOpacity
} from 'react-native'
import {
  InputArea,
  ImagePickerFunction,
  InputSelect,
  Loader
} from '../components'
import { ProgressSteps, ProgressStep } from 'react-native-progress-steps'
import { useNavigation } from '@react-navigation/native'
import { createRows } from '../utils/util'
import api from '../services/api'
import { FontAwesome } from 'react-native-vector-icons'
import Icon from '@expo/vector-icons/FontAwesome5'

import colors from '../constants/colors.json'
import states from '../constants/states.json'
import types from '../constants/types.json'

export default function AddProperty() {
  const navigation = useNavigation()

  const [loading, setLoading] = useState(false)
  const [data, setData] = useState({
    title: '',
    description: '',
    street: '',
    neighborhood: '',
    city: '',
    state: 'MG',
    country: 'Brasil',
    price: '',
    bedrooms: '',
    bathrooms: '',
    area: '',
    place: '',
    animal: true,
    type: 'Casa',
    images: []
  })
  const [activeStep, setActiveStep] = useState(0)

  const onChange = (type, value) => setData({ ...data, [type]: value })

  const onChangeImages = image => {
    setData({
      ...data,
      images: [...data.images, image]
    })
  }

  const addProperty = async () => {
    setLoading(true)

    const config = {
      headers: {
        'Content-Type': 'multipart/form-data'
      }
    }

    const formData = new FormData()

    formData.append('title', data.title)
    formData.append('description', data.description)
    formData.append('street', data.street)
    formData.append('neighborhood', data.neighborhood)
    formData.append('city', data.city)
    formData.append('state', data.state)
    formData.append('country', data.country)
    formData.append('price', parseFloat(data.price.replace(',', '.')))
    formData.append('bedrooms', parseInt(data.bedrooms, 10))
    formData.append('bathrooms', parseInt(data.bathrooms, 10))
    formData.append('area', parseFloat(data.area.replace(',', '.')))
    formData.append('place', parseInt(data.place, 10))
    formData.append('animal', data.animal)
    formData.append('type', data.type)

    data.images.forEach(image => {
      formData.append('files', image)
    })

    api
      .post('/user/property', formData, config)
      .then(res => {
        navigation.navigate('Anúnciar', { reload: true })
      })
      .catch(err => {
        console.error(err)
      })

    setLoading(false)
  }

  return (
    <ScrollView style={styles.container} showsVerticalScrollIndicator={false}>
      <Loader isLoading={loading} />

      <TouchableOpacity
        style={styles.backButton}
        onPress={() => navigation.navigate('Home')}
      >
        <Icon name={'arrow-left'} size={20} color={colors.yellow} />
      </TouchableOpacity>

      <KeyboardAvoidingView behavior="padding" enabled={Platform.OS === 'ios'}>
        <View style={styles.header}>
          <Text style={styles.headerMessage}>Vamos cadastrar seu imóvel</Text>
        </View>

        <ProgressSteps
          activeStep={activeStep}
          activeStepIconBorderColor={colors.yellow}
          completedProgressBarColor={colors.yellow}
          activeLabelColor={colors.yellow}
          completedLabelColor={colors.yellow}
          completedStepIconColor={colors.yellow}
          completedCheckColor={colors.platinum}
        >
          <ProgressStep
            key={0}
            testID="next-button-1"
            nextBtnText={'Próximo'}
            nextBtnStyle={styles.button}
            nextBtnTextStyle={styles.buttonText}
            scrollable={false}
            onNext={() => setActiveStep(activeStep + 1)}
          >
            <View style={styles.containerInput}>
              <Text style={styles.message}>
                Primeiramente, nos diga as informações que você gostaria de
                conter no seu anúncio
              </Text>
              <Text style={styles.label}>Título</Text>
              <InputArea
                testID="title-input"
                placeholder={'Título do anúncio...'}
                value={data.title}
                onChangeText={value => onChange('title', value)}
              />
              <Text style={styles.label}>Descrição</Text>
              <InputArea
                testID="description-input"
                placeholder={'Descrição do anúncio...'}
                value={data.description}
                multiline={true}
                textAlignVertical="top"
                style={styles.textArea}
                onChangeText={value => onChange('description', value)}
              />
            </View>
          </ProgressStep>

          <ProgressStep
            key={1}
            testID="next-button-2"
            nextBtnText={'Próximo'}
            previousBtnText={'Anterior'}
            nextBtnStyle={styles.button}
            previousBtnStyle={styles.button}
            nextBtnTextStyle={styles.buttonText}
            previousBtnTextStyle={styles.buttonText}
            scrollable={false}
            onNext={() => setActiveStep(activeStep + 1)}
            onPrevious={() => setActiveStep(activeStep - 1)}
          >
            <View style={styles.containerInput}>
              <Text style={styles.message}>
                Onde seu imóvel está localizado?
              </Text>
              <Text style={styles.label}>Rua</Text>
              <InputArea
                testID="street-input"
                placeholder={'Rua do imóvel...'}
                value={data.street}
                onChangeText={value => onChange('street', value)}
              />
              <Text style={styles.label}>Bairro</Text>
              <InputArea
                testID="neighborhood-input"
                placeholder={'Bairro do imóvel...'}
                value={data.neighborhood}
                onChangeText={value => onChange('neighborhood', value)}
              />
              <Text style={styles.label}>Cidade</Text>
              <InputArea
                testID="city-input"
                placeholder={'Cidade do imóvel...'}
                value={data.city}
                onChangeText={value => onChange('city', value)}
              />
              <Text style={styles.label}>Estado</Text>
              <InputSelect
                testID="state-input"
                items={states}
                selectedValue={data.state}
                onChange={item => onChange('state', item.value)}
              />
            </View>
          </ProgressStep>

          <ProgressStep
            key={2}
            testID="next-button-3"
            nextBtnText={'Próximo'}
            previousBtnText={'Anterior'}
            nextBtnStyle={styles.button}
            previousBtnStyle={styles.button}
            nextBtnTextStyle={styles.buttonText}
            previousBtnTextStyle={styles.buttonText}
            scrollable={false}
            onNext={() => setActiveStep(activeStep + 1)}
            onPrevious={() => setActiveStep(activeStep - 1)}
          >
            <View style={styles.containerInput}>
              <Text style={styles.message}>
                Precisamos de mais algumas informações do seu imóvel
              </Text>
              <Text style={styles.label}>Tipo</Text>
              <InputSelect
                testID="type-input"
                items={types}
                selectedValue={data.type}
                menuTitle="Qual o tipo de imóve?"
                onChange={item => onChange('type', item.value)}
              />
              <Text style={styles.label}>Quartos</Text>
              <InputArea
                testID="bedrooms-input"
                keyboardType={'number-pad'}
                placeholder={'Quantidade de quartos...'}
                value={data.bedrooms}
                onChangeText={value => onChange('bedrooms', value)}
              />
              <Text style={styles.label}>Banheiros</Text>
              <InputArea
                testID="bathrooms-input"
                keyboardType={'number-pad'}
                placeholder={'Quantidade de banheiros...'}
                value={data.bathrooms}
                onChangeText={value => onChange('bathrooms', value)}
              />
              <Text style={styles.label}>Area (m3)</Text>
              <InputArea
                testID="area-input"
                keyboardType={'number-pad'}
                placeholder={'Area do imóvel...'}
                value={data.area}
                onChangeText={value => onChange('area', value)}
              />
              <Text style={styles.label}>Vagas</Text>
              <InputArea
                testID="place-input"
                keyboardType={'number-pad'}
                placeholder={'É um imóvel para quantas pessoas?'}
                value={data.place}
                onChangeText={value => onChange('place', value)}
              />
              <Text style={styles.label}>Animal</Text>
              <InputSelect
                testID="animal-input"
                items={[
                  { label: 'Sim', value: true },
                  { label: 'Não', value: false }
                ]}
                selectedValue={data.animal}
                menuTitle="Seu imóvel pode conter animais?"
                onChange={item => onChange('animal', item.value)}
              />
            </View>
          </ProgressStep>

          <ProgressStep
            key={3}
            testID="next-button-4"
            nextBtnText={'Próximo'}
            previousBtnText={'Anterior'}
            nextBtnStyle={styles.button}
            previousBtnStyle={styles.button}
            nextBtnTextStyle={styles.buttonText}
            previousBtnTextStyle={styles.buttonText}
            scrollable={false}
            onNext={() => setActiveStep(activeStep + 1)}
            onPrevious={() => setActiveStep(activeStep - 1)}
          >
            <View style={styles.containerInput}>
              <Text style={styles.message}>
                Selecione algumas imagens do seu imóvel para deixa-lo mais
                apresentável
              </Text>
              <SafeAreaView style={styles.imageContainer}>
                <FlatList
                  data={createRows([...data.images, { imagePicker: true }], 2)}
                  keyExtractor={(item, index) => index.toString()}
                  numColumns={2}
                  scrollEnabled={false}
                  renderItem={({ item }) => {
                    if (item.empty) {
                      return (
                        <View
                          style={{
                            ...styles.image,
                            backgroundColor: 'transparent'
                          }}
                        />
                      )
                    }

                    if (item.imagePicker) {
                      return (
                        <ImagePickerFunction
                          style={styles.image}
                          onChange={image => onChangeImages(image)}
                        >
                          <View style={styles.addIcon}>
                            <FontAwesome
                              name="plus"
                              color={colors['blue-secondary']}
                              size={20}
                            />
                          </View>
                        </ImagePickerFunction>
                      )
                    }

                    return (
                      <Image source={{ uri: item.uri }} style={styles.image} />
                    )
                  }}
                />
              </SafeAreaView>
            </View>
          </ProgressStep>

          <ProgressStep
            key={4}
            testID="submit-button"
            previousBtnText={'Anterior'}
            finishBtnText={'Finalizar'}
            nextBtnStyle={styles.button}
            previousBtnStyle={styles.button}
            nextBtnTextStyle={styles.buttonText}
            previousBtnTextStyle={styles.buttonText}
            scrollable={false}
            onSubmit={() => addProperty()}
            onPrevious={() => setActiveStep(activeStep - 1)}
          >
            <View style={styles.containerInput}>
              <Text style={styles.message}>
                Vamos definir um preço para seu imóvel
              </Text>
              <Text style={styles.label}>Preço</Text>
              <InputArea
                testID="price-input"
                keyboardType={'number-pad'}
                label="R$"
                placeholder={'0,00'}
                value={data.price}
                onChangeText={value => onChange('price', value)}
              />
            </View>
          </ProgressStep>
        </ProgressSteps>
      </KeyboardAvoidingView>
    </ScrollView>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.platinum,
    padding: 30
  },

  containerInput: {
    marginBottom: 20
  },

  textArea: {
    height: 100,
    justifyContent: 'flex-start'
  },

  header: {
    marginTop: 25,
    justifyContent: 'center',
    alignItems: 'center'
  },

  headerMessage: {
    width: 300,
    fontSize: 23,
    fontWeight: '600',
    color: colors.blue,
    textAlign: 'left'
  },

  message: {
    marginTop: 5,
    fontSize: 15,
    fontWeight: '500',
    color: '#999',
    textAlign: 'center',
    marginBottom: 10
  },

  label: {
    marginTop: 15,
    marginBottom: -10,
    marginLeft: 20,
    fontSize: 14,
    fontWeight: '500',
    color: colors.blue
  },

  button: {
    height: 35,
    width: 90,
    backgroundColor: colors.yellow,
    borderRadius: 24,
    alignItems: 'center',
    justifyContent: 'center'
  },

  buttonText: {
    color: colors.blue,
    fontWeight: 'bold',
    fontSize: 16
  },

  imageContainer: {
    flex: 1,
    marginTop: 10
  },

  image: {
    flexGrow: 1,
    flexBasis: 0,
    backgroundColor: '#FFFFFF',
    borderRadius: 10,
    margin: 3,
    paddingHorizontal: 8,
    paddingVertical: 5,
    justifyContent: 'center',
    width: '100%',
    height: 170
  },

  addIcon: {
    alignSelf: 'center'
  }
})
