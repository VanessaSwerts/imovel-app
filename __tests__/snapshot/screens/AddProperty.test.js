import React from 'react'
import { create } from 'react-test-renderer'
import AddProperty from '../../../src/screens/AddProperty'

jest.runAllTimers()
jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      navigate: jest.fn()
    })
  }
})
jest.mock('../../../src/contexts/loading', () => {
  return {
    ...jest.requireActual('../../../src/contexts/loading'),
    useLoading: () => ({
      startLoading: jest.fn(),
      stopLoading: jest.fn(),
      loading: jest.fn()
    })
  }
})

describe('AddProperty snapshot test', () => {
  it('render AddProperty screen correctly', () => {
    const tree = create(<AddProperty />)
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
