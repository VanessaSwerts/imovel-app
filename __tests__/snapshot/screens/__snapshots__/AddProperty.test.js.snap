// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProperty snapshot test render AddProperty screen correctly 1`] = `
<RCTScrollView
  showsVerticalScrollIndicator={false}
  style={
    Object {
      "backgroundColor": "#F0F0F0",
      "flex": 1,
      "paddingHorizontal": 30,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 30,
          "marginTop": 20,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 30,
            "justifyContent": "center",
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "width": 30,
          }
        }
      >
        <Icon
          color="#2C3E50"
          name="arrow-left"
          size={20}
        />
      </View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#2C3E50",
            "fontSize": 23,
            "fontWeight": "600",
            "opacity": 0.8,
            "textAlign": "center",
          }
        }
      >
        Novo anuncio
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "paddingBottom": 0,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#2C3E50",
            "fontSize": 23,
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        Vamos cadastrar seu im√≥vel
      </Text>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "flexDirection": "row",
              "justifyContent": "space-evenly",
              "marginBottom": 50,
              "position": "relative",
              "top": 30,
            }
          }
        >
          <View>
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "column",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "borderColor": "#0565FC",
                      "borderRadius": 20,
                      "borderWidth": 5,
                      "bottom": 2,
                      "height": 40,
                      "width": 40,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "alignSelf": "center",
                        "top": 6.666666666666667,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "black",
                        }
                      }
                    >
                      1
                    </Text>
                  </Text>
                </View>
                <Text
                  style={
                    Object {
                      "color": "#0565FC",
                      "flexWrap": "wrap",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "paddingTop": 4,
                      "textAlign": "center",
                      "width": 100,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "borderTopColor": "#ebebe4",
                      "borderTopStyle": "solid",
                      "borderTopWidth": 3,
                      "left": 48,
                      "marginLeft": 22,
                      "position": "absolute",
                      "right": 0,
                      "top": 18.018018018018015,
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View>
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "column",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#ebebe4",
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "alignSelf": "center",
                        "top": 9,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "white",
                        }
                      }
                    >
                      2
                    </Text>
                  </Text>
                </View>
                <Text
                  style={
                    Object {
                      "color": "lightgray",
                      "flexWrap": "wrap",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "marginTop": 4,
                      "paddingTop": 4,
                      "textAlign": "center",
                      "width": 100,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "borderTopColor": "#ebebe4",
                      "borderTopStyle": "solid",
                      "borderTopWidth": 3,
                      "left": 0,
                      "marginRight": 22,
                      "position": "absolute",
                      "right": 44,
                      "top": 18,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "borderTopColor": "#ebebe4",
                      "borderTopStyle": "solid",
                      "borderTopWidth": 3,
                      "left": 44,
                      "marginLeft": 22,
                      "position": "absolute",
                      "right": 0,
                      "top": 18,
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View>
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "column",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#ebebe4",
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "alignSelf": "center",
                        "top": 9,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "white",
                        }
                      }
                    >
                      3
                    </Text>
                  </Text>
                </View>
                <Text
                  style={
                    Object {
                      "color": "lightgray",
                      "flexWrap": "wrap",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "marginTop": 4,
                      "paddingTop": 4,
                      "textAlign": "center",
                      "width": 100,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "borderTopColor": "#ebebe4",
                      "borderTopStyle": "solid",
                      "borderTopWidth": 3,
                      "left": 0,
                      "marginRight": 22,
                      "position": "absolute",
                      "right": 44,
                      "top": 18,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "borderTopColor": "#ebebe4",
                      "borderTopStyle": "solid",
                      "borderTopWidth": 3,
                      "left": 44,
                      "marginLeft": 22,
                      "position": "absolute",
                      "right": 0,
                      "top": 18,
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View>
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "column",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#ebebe4",
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "alignSelf": "center",
                        "top": 9,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "white",
                        }
                      }
                    >
                      4
                    </Text>
                  </Text>
                </View>
                <Text
                  style={
                    Object {
                      "color": "lightgray",
                      "flexWrap": "wrap",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "marginTop": 4,
                      "paddingTop": 4,
                      "textAlign": "center",
                      "width": 100,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "borderTopColor": "#ebebe4",
                      "borderTopStyle": "solid",
                      "borderTopWidth": 3,
                      "left": 0,
                      "marginRight": 22,
                      "position": "absolute",
                      "right": 44,
                      "top": 18,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "borderTopColor": "#ebebe4",
                      "borderTopStyle": "solid",
                      "borderTopWidth": 3,
                      "left": 44,
                      "marginLeft": 22,
                      "position": "absolute",
                      "right": 0,
                      "top": 18,
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View>
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "column",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#ebebe4",
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "alignSelf": "center",
                        "top": 9,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "white",
                        }
                      }
                    >
                      5
                    </Text>
                  </Text>
                </View>
                <Text
                  style={
                    Object {
                      "color": "lightgray",
                      "flexWrap": "wrap",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "marginTop": 4,
                      "paddingTop": 4,
                      "textAlign": "center",
                      "width": 100,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "borderTopColor": "#ebebe4",
                      "borderTopStyle": "solid",
                      "borderTopWidth": 3,
                      "left": 0,
                      "marginRight": 22,
                      "position": "absolute",
                      "right": 44,
                      "top": 18,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "marginBottom": 20,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#7F8C8D",
                      "fontSize": 15,
                      "fontWeight": "500",
                      "marginBottom": 10,
                      "marginTop": 5,
                      "textAlign": "center",
                    }
                  }
                >
                  Primeiramente, nos diga as informa√ß√µes que voc√™ gostaria de conter no seu an√∫ncio
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#7F8C8D",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "marginBottom": -10,
                      "marginLeft": 20,
                      "marginTop": 15,
                    }
                  }
                >
                  T√≠tulo
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "alignSelf": "stretch",
                        "backgroundColor": "#FFFFFF",
                        "borderColor": "#0565FC",
                        "borderRadius": 25,
                        "borderWidth": 1,
                        "elevation": 2,
                        "flexDirection": "row",
                        "marginTop": 15,
                        "paddingHorizontal": 12,
                        "shadowOffset": Object {
                          "height": 0,
                          "width": 0,
                        },
                        "shadowOpacity": 0.2,
                        "shadowRadius": 2,
                      },
                      undefined,
                    ]
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder="T√≠tulo do an√∫ncio..."
                    placeholderTextColor="#AAADB3"
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    style={
                      Array [
                        Object {
                          "color": "#2C3E50",
                          "height": 46,
                          "paddingHorizontal": 8,
                          "paddingVertical": 10,
                          "width": "100%",
                        },
                        undefined,
                      ]
                    }
                    testID="title-input"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
                <Text
                  style={
                    Object {
                      "color": "#7F8C8D",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "marginBottom": -10,
                      "marginLeft": 20,
                      "marginTop": 15,
                    }
                  }
                >
                  Descri√ß√£o
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "alignSelf": "stretch",
                        "backgroundColor": "#FFFFFF",
                        "borderColor": "#0565FC",
                        "borderRadius": 25,
                        "borderWidth": 1,
                        "elevation": 2,
                        "flexDirection": "row",
                        "marginTop": 15,
                        "paddingHorizontal": 12,
                        "shadowOffset": Object {
                          "height": 0,
                          "width": 0,
                        },
                        "shadowOpacity": 0.2,
                        "shadowRadius": 2,
                      },
                      undefined,
                    ]
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    multiline={true}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder="Descri√ß√£o do an√∫ncio..."
                    placeholderTextColor="#AAADB3"
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    style={
                      Array [
                        Object {
                          "color": "#2C3E50",
                          "height": 46,
                          "paddingHorizontal": 8,
                          "paddingVertical": 10,
                          "width": "100%",
                        },
                        Object {
                          "height": 100,
                          "justifyContent": "flex-start",
                        },
                      ]
                    }
                    testID="description-input"
                    textAlignVertical="top"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "marginTop": 90,
                }
              }
            >
              <View
                style={
                  Object {
                    "bottom": 40,
                    "left": 60,
                    "position": "absolute",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                      "padding": 8,
                      "textAlign": "center",
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#007AFF",
                          "fontSize": 18,
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "bottom": 40,
                    "position": "absolute",
                    "right": 60,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#0565FC",
                      "borderRadius": 24,
                      "height": 35,
                      "justifyContent": "center",
                      "opacity": 1,
                      "padding": 8,
                      "textAlign": "center",
                      "width": 90,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "fontSize": 16,
                          "fontWeight": "bold",
                        },
                      ]
                    }
                  >
                    Pr√≥ximo
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
